### DISCUSSION TITLE
Nextjs api route proxy logger

### DISCUSSION THREAD

**Discussion by popuguytheparrot** [chimurai/http-proxy-middleware#897] [Q&A] [answered]
Hi. How to use with nextjs and logger all request and errors?

```js

// pages/api/gateway/[[...slug]].ts
import winston, {format, transports} from "winston";

export const logger = winston.createLogger({
  format: format.combine(format.splat(), format.simple()),
  transports: [new transports.Console()],
});

const proxy = createProxyMiddleware({
  target,
  changeOrigin: true,
  pathRewrite: { "^/api/": "" },
  secure: false,
  preserveHeaderKeyCase: true,
  xfwd: true,
  logger: logger,
});

export default (req, res) => {
 // what here???
}

```

**Comment by chimurai** [comment, marked as answer]
`logger` is a v3@beta functionality: https://github.com/chimurai/http-proxy-middleware/discussions/768.
Make sure to have the correct version installed.

You can find some examples here:
NextJS: https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/servers.md#nextjs
log: https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/response-interceptor.md#log-request-and-response
